### ISSUE TITLE
SQLite auto-detection using 32-bit overflow fails on 64-bit

### ISSUE THREAD

**Issue by gotoalberto** [bitcoin-abe/bitcoin-abe#168]
ABE doesn't work using SQLite as database.
I'm using ABE to show information about a custom Bitcoin testnet

This is my abe.conf:

```
datadir += [{
	"dirname": "/var/lib/bitcoind/1/testnet3",
        "chain":"test",
        "policy":"Bitcoin",
        "address_version":"\u006f"
}]
host 0.0.0.0
port 8085
dbtype sqlite3
default-loader blkfile
connect-args /var/lib/abe/db.sqlite
```

It works normally, indexing blocks until it arrives to this block:
```
Failed to catch up {'blkfile_offset': 1440485, 'blkfile_number': 100000, 'chain_id': 16, 'loader': None, 'conf': u'/etc/bitcoind/1.conf', 'dirname': u'/var/lib/bitcoind/1/testnet3', 'id': 8}
Traceback (most recent call last):
  File "/usr/local/lib/abe/env/local/lib/python2.7/site-packages/Abe/DataStore.py", line 2517, in catch_up
    if not store.catch_up_rpc(dircfg):
  File "/usr/local/lib/abe/env/local/lib/python2.7/site-packages/Abe/DataStore.py", line 2692, in catch_up_rpc
    store.import_block(block, chain = chain)
  File "/usr/local/lib/abe/env/local/lib/python2.7/site-packages/Abe/DataStore.py", line 1139, in import_block
    len(b['transactions']), b['search_block_id']))
  File "/usr/local/lib/abe/env/local/lib/python2.7/site-packages/Abe/DataStore.py", line 251, in sql
    store._sql.sql(stmt, params)
  File "/usr/local/lib/abe/env/local/lib/python2.7/site-packages/Abe/SqlAbstraction.py", line 457, in sql
    sql._execute(cached, params)
  File "/usr/local/lib/abe/env/local/lib/python2.7/site-packages/Abe/SqlAbstraction.py", line 438, in _execute
    sql.cursor().execute(stmt, params)
OverflowError: Python int too large to convert to SQLite INTEGER
Abe initialized.
Listening on http://0.0.0.0:8085
```

After this block is reached in the network, ABE don't index blocks anymore.

**Comment by dermoth** [comment]
It looks like the SQLIte auto-detection using 32-bit overflow fails on 64-bit.

I can change it to a 64bit overflow, but i'd like to have someone with a 32bit system to test it. As an alternative we can auto-detect the int size in Python and select the right test number.

@jtobey , do you have any preferences?

Also, an additional related issue is that the ```--int-type``` option is being ignored, and this prevent from forcing the proper behavior. See Issue #178.

**Comment by dermoth** [comment]
It turns out the issue wasn't really the same, but it fails trying to insert very large numbers. I added one such number to the test inserts to catch this as well; although it's not the same issue it's the same fix.

The fix is in PR #179

I need to check MySQL and PgSQL are not affected by this change before merging.

**Comment by dermoth** [comment]
Please test now with the latest code, it should be fixed.

Note that sqlite is pretty limited in functionality. You can only have one Abe process using the database, and that process won't be able to serve requests while loading up blocks. For better concurrency you should think about using a real database such as MySQL/MariaDB or PostgreSQL.

**Comment by dermoth** [comment]
Please test now with the latest code, it should be fixed in PR #179.

Note that sqlite is pretty limited in functionality. You can only have one Abe process using the database, and that process won't be able to serve requests while loading up blocks. For better concurrency you should think about using a real database such as MySQL/MariaDB or PostgreSQL.
